lib/data_freshness/main.py

```diff
@@ -0,0 +1,137 @@
+                user_uuid=user_uuid,
+                institution_id=inst.institution_id,
+                utc_starttime=utc_starttime,
+                utc_endtime=utc_endtime,
+                page=1,
+            )
+        )
+        balance_inst_coroutines.append(
+            api_client.balances.get(
+                user_uuid=user_uuid,
+                institution_id=inst.institution_id,
+                utc_starttime=utc_starttime,
+                utc_endtime=utc_endtime,
+                page=1,
+            )
+        )
+        alerts_inst_coroutines.append(
+            api_client.balances.get(
+                user_uuid=user_uuid,
+                institution_id=inst.institution_id,
+                utc_starttime=utc_starttime,
+                utc_endtime=utc_endtime,
+                page=1,
+            )
+        )
+
+    transactions_per_institution = await asyncio.gather(*transaction_inst_coroutine)
+    balances_per_institution = await asyncio.gather(*balance_inst_coroutines)
+    alerts_per_institution = await asyncio.gather(*alerts_inst_coroutines)
+
+
+    # STEP 3: flatten all balances, transactions, and alerts from all institutions, and get most recent time
+    transactions_flattened = []
+    for ix, transactions in enumerate(transactions_per_institution):
+        institution_id = institutions[ix].institution_id
+        for transaction in transactions:
+            transactions_flattened.append(
+                dict(transaction, institution_id=institution_id))
+
+    balances_flattened = []
+    for ix, balances in enumerate(balances_per_institution):
+        institution_id = institutions[ix].institution_id
+        for balance in balances:
+            balances_flattened.append(dict(balance, institution_id=institution_id))
+    
+    alerts_flattened = []
+    for ix, alerts in enumerate(alerts_per_institution):
+        institution_id = institutions[ix].institution_id
+        for alert in alerts:
+            alerts_flattened.append(dict(alert, institution_id=institution_id))
+
+
+    transactions_sorted = sorted(transactions_flattened, key=lambda x: x["ts"], reverse=True)
+    balances_sorted = sorted(balances_flattened, key=lambda x: x["ts"], reverse=True)
+    alerts_sorted = sorted(alerts_flattened, key=lambda x: x["ts"], reverse=True)
+
+
+    if not transactions_sorted:
+        transactions_ts_recent = 0
+    else:
+        transactions_ts_recent = transactions_sorted[0]["ts"]
+        
+    if not balances_sorted:
+        balances_ts_recent = 0
+    else:
+        balances_ts_recent = balances_sorted[0]["ts"]
+
+    if not alerts_sorted:
+        alerts_ts_recent = 0
+    else:
+        alerts_ts_recent = alerts_sorted[0]["ts"]
+
+    most_recent_ts = max(transactions_ts_recent, balances_ts_recent, alerts_ts_recent)
+    if most_recent_ts is None:
+        data_freshness = None
+    else:
+        data_freshness = round((utc_endtime.timestamp() - most_recent_ts) / 60)
+
+    return data_freshness
```

Comment:
Here, you could slightly reformat to completely do away with `else:` ->
```suggestion
    if most_recent_ts == 0:
       return None
    
    data_freshness = round((utc_endtime.timestamp() - most_recent_ts) / 60)
    return data_freshness
```